### PR TITLE
STREAMS-1365 : add security guide section

### PR DESCRIPTION
**Pull request by dominiquemongelli**
Thank you for your contribution to the Axway Streams documentation.

## Describe the changes

Add security guide section

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

_Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your change._

## Checklist for approvers

_This section is to be filled out by project maintainers only._

Before approving this PR, please make sure it:

* [ ] Does not have conflicts with the base branch
* [ ] Is believed to be accurate (technical accuracy to be checked with an SME for PRs originating from outside R&D)
* [ ] Follows [Axway style](https://techweb.axway.com/confluence/display/RIE/Style+guide)
* [ ] Follows [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)
* [ ] Follows [best practices](https://axway-open-docs.netlify.com/docs/contribution_guidelines/bestpracticedevdoc/)
* [ ] Does not contain typos, grammatical errors, etc.
* [ ] Does not expose any sensitive information
* [ ] Passes the Markdown linter rules (automated via CI check)
* [ ] Does not contain broken links

### CODE REVIEW

File: content/en/docs/streams/security/security_best_practices.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Internet access limitation
+
+As much as possible, limit the number of Internet access points. Do not open useless Internet connections and limit interconnections with external networks as much as possible. This limits the product’s attack surface, reduces the risk of external attacks, and makes it easier to audit the product.
+
+For a list of default ports that are opened by the API Gateway components, see [Default ports](/docs/apim_installation/apigtw_install/system_requirements/#default-ports).
+
+## Password policy
+
+In line with security best practices, you can configure a password policy for administrator users in API Gateway Manager. Password policy refers to the size and complexity of the password, as well as to all the rules to manage the password.
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_best_practices.md:26]: Remove Reference to API Gateway Manager

File: content/en/docs/streams/security/security_features.md [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+* All outbound connections made by Streams to destination endpoints are verified by default to ensure that a trusted certificate is used. 
+* Connections between Streams microservice and database is SSL-secured by default. 
+* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured. 
 
-## Secure storage
+## Encrypted data-at-rest
 
-* Streams Database columns likely to contain sensitive authentication information to external component (e.g. Publisher configuration, Headers values) are stored encrypted in database.
-* All sensitive Streams configuration items (e.g. MariaDB passwords) are stored in [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) by default which can be used in conjunction with a [Key Management Service](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/) (KMS) to enable secret data encryption.
+* Kubernetes secret data can be encrypted at rest. Raw encryption key can be stored within a KMS provider
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:20]: Kubernetes secret**s**

File: content/en/docs/streams/security/security_features.md [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+* Connections between Streams microservice and database is SSL-secured by default. 
+* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured. 
 
-## Secure storage
+## Encrypted data-at-rest
 
-* Streams Database columns likely to contain sensitive authentication information to external component (e.g. Publisher configuration, Headers values) are stored encrypted in database.
-* All sensitive Streams configuration items (e.g. MariaDB passwords) are stored in [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) by default which can be used in conjunction with a [Key Management Service](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/) (KMS) to enable secret data encryption.
+* Kubernetes secret data can be encrypted at rest. Raw encryption key can be stored within a KMS provider 
+* Data-at-rest is enabled by default for mariadb
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:21]: mariadb => MariaDB

File: content/en/docs/streams/security/security_features.md [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
+
+## Password management
+
+* Passwords for hazelcast and mariadb and are managed via kubernetes secrets
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:25]: MariaDB

File: content/en/docs/streams/security/security_features.md [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
 
-* Streams Database columns likely to contain sensitive authentication information to external component (e.g. Publisher configuration, Headers values) are stored encrypted in database.
-* All sensitive Streams configuration items (e.g. MariaDB passwords) are stored in [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) by default which can be used in conjunction with a [Key Management Service](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/) (KMS) to enable secret data encryption.
+* Kubernetes secret data can be encrypted at rest. Raw encryption key can be stored within a KMS provider 
+* Data-at-rest is enabled by default for mariadb
+
+## Password management
+
+* Passwords for hazelcast and mariadb and are managed via kubernetes secrets  
+* A keyfile is required to enable data-at-rest encryption on mariadb
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:26]: MariaDB

File: content/en/docs/streams/security/security_features.md [hunk 6]
```diff
@@ -1,22 +1,32 @@
...
+
+## Certificate management
+ 
+* A CA certificate, server certificate and a server key are required for TLS communication between Streams and the mariadb
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:30]: MariaDB

File: content/en/docs/streams/security/security_features.md [hunk 7]
```diff
@@ -1,22 +1,32 @@
...
+* SSL/TLS certificate is required for TLS communication on the ingress
+* Certificates are stored and managed via kubernetes secrets
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:32]: **K**ubernetes

File: content/en/docs/streams/security/security_features.md [hunk 8]
```diff
@@ -1,22 +1,32 @@
...
+The following secure connections are available: 
 
-* All inbound connections to Streams are SSL-secured by default.
-* All outbound connections made by Streams to destination endpoints can be verified to ensure that a trusted certificate is used.
-* Connections between Streams microservice and database can be SSL-secured.
-* Connections between Streams microservice and Kafka can be SSL-secured.
-* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured.
+* All inbound connections to Streams are SSL-secured by default. 
+* All outbound connections made by Streams to destination endpoints are verified by default to ensure that a trusted certificate is used. 
+* Connections between Streams microservice and database is SSL-secured by default.
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:15]: microservice**s**

File: content/en/docs/streams/security/security_architecture.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+## Streams architecture
+
+The following diagram shows the product architecture from a security perspective.  The legend explains the security level on connections (SSL by default, always SSL, can be SSL, and so on) and on data storage (signed or encrypted).
+
+![Streams architecture and connection](/Images/security/sec_arch.png)
+
+The diagram includes the following components:
+
+* MariaDB: RDBMS use to persist topics and subscriptions
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_architecture.md:17]: used

File: content/en/docs/streams/security/security_architecture.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+"title": "Security architecture",
+"linkTitle": "Security architecture",
+"weight": 30,
+"date": "2020-10-08",
+"description": "Architecture of Streams from a security perspective."
+}
+
+## Streams architecture
+
+The following diagram shows the product architecture from a security perspective.  The legend explains the security level on connections (SSL by default, always SSL, can be SSL, and so on) and on data storage (signed or encrypted).
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_architecture.md:11]: SSL => TLS ?

File: content/en/docs/streams/security/security_best_practices.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+* Anyone can generate his own certificate and you need a very secure process to receive/send these certificates to make sure they are coming from the right partner. When using CA-signed certificates, you can rely on the CA.
+* If self-signed certificates are not securely stored, anyone can change them. CA-signed certificates also must be securely stored, but no one can change them.
+* There is no way to revoke self-signed certificates.
+
+## Internet access limitation
+
+As much as possible, limit the number of Internet access points. Do not open useless Internet connections and limit interconnections with external networks as much as possible. This limits the product’s attack surface, reduces the risk of external attacks, and makes it easier to audit the product.
+
+For a list of default ports that are opened by the API Gateway components, see [Default ports](/docs/apim_installation/apigtw_install/system_requirements/#default-ports).
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_best_practices.md:22]: Link to Streams doc instead

File: content/en/docs/streams/security/security_best_practices.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+For more information on setting the password policy for administrator users, see the
+[Configure a password policy for admin users](/docs/apim_administration/apigtw_admin/manage_user_access/#configure-a-password-policy-for-admin-users).
+
+You can also configure password policies for [API Manager](/docs/apim_administration/apimgr_admin/api_mgmt_admin/#enforce-password-changes) and [API Portal](/docs/apim_administration/apiportal_admin/customize_page_content/#enforce-password-policies) users.
+
+## Sensitive files and databases
+
+In general, it is good practice to limit the number of administrators with access to the product.
+
+Ensure the default protection mechanisms on sensitive files used by API Gateway and API Portal remain in place.  For example, the product’s files are installed with read, update, and delete privileges such that only the product can access them by default.  There should be no reason to change these privileges.
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_best_practices.md:39]: Remove ref to API Gateway & API Portal

File: content/en/docs/streams/security/security_best_practices.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_best_practices.md:40]: Not relevant for Streams

File: content/en/docs/streams/security/security_features.md [hunk 14]
```diff
@@ -1,22 +1,32 @@
...
-weight: 30
+weight: 20
 date: 2020-09-18
 description: Summary of the main security features of Streams.
 ---
 
 ## Secure connections
 
-The following secure connections are available:
+The following secure connections are available:
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:11]: Trailing whitespace cause markdown lint issue

File: content/en/docs/streams/security/security_features.md [hunk 15]
```diff
@@ -1,22 +1,32 @@
...
 
-The following secure connections are available:
+The following secure connections are available: 
 
-* All inbound connections to Streams are SSL-secured by default.
-* All outbound connections made by Streams to destination endpoints can be verified to ensure that a trusted certificate is used.
-* Connections between Streams microservice and database can be SSL-secured.
-* Connections between Streams microservice and Kafka can be SSL-secured.
-* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured.
+* All inbound connections to Streams are SSL-secured by default.
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:13]: SSL => TLS ?

File: content/en/docs/streams/security/security_features.md [hunk 16]
```diff
@@ -1,22 +1,32 @@
...
 
-* All inbound connections to Streams are SSL-secured by default.
-* All outbound connections made by Streams to destination endpoints can be verified to ensure that a trusted certificate is used.
-* Connections between Streams microservice and database can be SSL-secured.
-* Connections between Streams microservice and Kafka can be SSL-secured.
-* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured.
+* All inbound connections to Streams are SSL-secured by default. 
+* All outbound connections made by Streams to destination endpoints are verified by default to ensure that a trusted certificate is used. 
+* Connections between Streams microservice and database is SSL-secured by default. 
+* Connections to other Axway products (for example, Axway API Gateway) can be SSL-secured.
```
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:16]: SSL => TLS ?

File: content/en/docs/streams/security/security_features.md (hunk 2 above)
Comment (allandenis38) [inline content/en/docs/streams/security/security_features.md:20]: Missing "." at end of sentence

**Review by allandenis38 — CHANGES_REQUESTED** [review]

File: content/en/docs/streams/security/security_best_practices.md (hunk 1 above)
Comment (dominiquemongelli) [reply content/en/docs/streams/security/security_best_practices.md:26]: I've removed the section as the password policy cannot be configured on Streams.

File: content/en/docs/streams/security/security_features.md [hunk 17]
```diff
@@ -10,13 +10,23 @@ description: Summary of the main security features of Streams.
...
+* Connections between Streams microservice and database is TLS-secured by default.
+* Connections to other Axway products (for example, Axway API Gateway) can be TLS-secured.
 
-## Secure storage
+## Encrypted data-at-rest
 
-* Streams Database columns likely to contain sensitive authentication information to external component (e.g. Publisher configuration, Headers values) are stored encrypted in database.
-* All sensitive Streams configuration items (e.g. MariaDB passwords) are stored in [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) by default which can be used in conjunction with a [Key Management Service](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/) (KMS) to enable secret data encryption.
+* Kubernetes secrets data can be encrypted at rest. Raw encryption key can be stored within a KMS provider.
+* Data-at-rest is enabled by default for MariaDB.
```
Comment (benjaminranc) [inline content/en/docs/streams/security/security_features.md:21]:
```suggestion
* Encrypted data-at-rest is enabled by default for MariaDB.
```

**Pull request closed without merging**